cloud/deployment/inspect/inspect.go

```diff
@@ -0,0 +1,157 @@
+	WorkspaceID    string    `mapstructure:"workspace-id"`
+	ClusterID      string    `mapstructure:"cluster-id"`
+	AirflowVersion string    `mapstructure:"airflow-version"`
+	ReleaseName    string    `mapstructure:"release-name"`
+	DeploymentURL  string    `mapstructure:"deployment-url"`
+	WebserverURL   string    `mapstructure:"webserver-url"`
+	UpdatedAt      time.Time `mapstructure:"updated-at"`
+	Status         string    `mapstructure:"status"`
+}
+
+type deploymentConfig struct {
+	Name              string              `mapstructure:"name"`
+	Description       string              `mapstructure:"description"`
+	ClusterID         string              `mapstructure:"cluster-id"` // this is also in deploymentInfo
+	RunTimeVersion    string              `mapstructure:"runtime-version"`
+	SchedulerAU       int                 `mapstructure:"scheduler-au"`
+	SchedulerReplicas int                 `mapstructure:"scheduler-replicas"`
+	AlertEmails       []string            `mapstructure:"alert-emails"`
+	WorkerQueues      []astro.WorkerQueue `mapstructure:"worker-queues"`
+	AstroVariables    Variables           `mapstructure:"astronomer-variables"`
+}
+
+type PrintableDeployment struct {
+	DeploymentInfo   map[string]interface{} `mapstructure:"deployment-info"`
+	DeploymentConfig map[string]interface{} `mapstructure:"deployment-config"`
+}
+
+type Variables struct {
+	EnvFromFile  string
+	VariableList []astro.EnvironmentVariablesObject `mapstructure:"astronomer-variables"`
+}
+
+var (
+	jsonMarshal = json.MarshalIndent
+	yamlMarshal = yaml.Marshal
+)
+
+const (
+	jsonFormat = "json"
+)
+
+func Inspect(wsID, deploymentName, deploymentID, outputFormat string, client astro.Client, out io.Writer) error {
+	var (
+		requestedDeployment                    astro.Deployment
+		err                                    error
+		infoToPrint                            []byte
+		deploymentInfoMap, deploymentConfigMap map[string]interface{}
+	)
+	// get or select the deployment
+	requestedDeployment, err = deployment.GetDeployment(wsID, deploymentID, deploymentName, client)
+	if err != nil {
+		return err
+	}
+
+	deploymentInfoMap, err = getDeploymentInspectInfo(&requestedDeployment)
+	if err != nil {
+		return err
+	}
+
+	deploymentConfigMap, err = getDeploymentConfig(&requestedDeployment)
+	if err != nil {
+		return err
+	}
+
+	infoToPrint, err = formatPrintableDeployment(deploymentInfoMap, deploymentConfigMap, outputFormat)
+	if err != nil {
+		return err
+	}
+	fmt.Fprint(out, string(infoToPrint)+"\n")
+	return nil
+}
+
+func getDeploymentInspectInfo(sourceDeployment *astro.Deployment) (map[string]interface{}, error) {
+	var (
+		deploymentURL string
+		c             config.Context
+		err           error
+	)
+	c, err = config.GetCurrentContext()
+	if err != nil {
+		return nil, err
+	}
+
+	deploymentURL = "cloud." + c.Domain + "/" + sourceDeployment.Workspace.ID + "/deployments/" + sourceDeployment.ID + "/analytics"
```

Comment:
We have this url construction at multiple places in code. Can we move it to a function and re-use?

Reply:
☝️

Reply:
yes I was questioning when to start that so now is as good a time as any :)